File: MusicShop.Model/Requests/AmplifierUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicShop.Model.Requests
{
    public class AmplifierUpdateRequest
    {
        public int? BrandId { get; set; }
        public string? Model { get; set; }
        public decimal? Price { get; set; }
        public string? Description { get; set; }
        public int? Voltage { get; set; }
        public int? PowerRating { get; set; }
        public bool? HeadphoneJack { get; set; }
        public bool? Usbjack { get; set; }
        public int? NumberOfPresets { get; set; }
    }
}

File: UI/musicshop_admin/lib/models/amplifier/amplifier_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'amplifier_update_request.g.dart';

@JsonSerializable()
class AmplifierUpdateRequest {
  int? id;
  int? brandId;
  String? model;
  double? price;
  String? description;
  int? voltage;
  int? powerRating;
  bool? headphoneJack;
  bool? usbjack;
  int? numberOfPresets;
  String? productImage;

  AmplifierUpdateRequest();

  factory AmplifierUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$AmplifierUpdateRequestFromJson(json);

  Map<String, dynamic> toJson() => _$AmplifierUpdateRequestToJson(this);
}
